### DISCUSSION TITLE
Using Github api to create a file

### DISCUSSION THREAD

**Discussion by Karan-S-dev** [community/community#28008] [Apps, API and Webhooks] [answered]
Hi,

 I am new to github api and trying to create a file in my repo using api.
Iam using postman to check the api first.

PUT Method:

URL:https://api.github.com/repos/KaranS-hexaware/Rapidx_Documentation/contents/text.txt

Body:{
   "message":"my commit message",
   "committer":{
      "name":"Karan S",
      "email":"1000055393@hexaware.com"
   },
   "content":"bXkgbmV3IGZpbGUgY29udGVudHM="
}

Header:Authorization:Bearer ****************************************************
             Content-Type:application/vnd.api+json

The response is:
{
    "message": "Not Found",
    "documentation_url": "https://docs.github.com/rest/reference/repos#create-or-update-file-contents"
}

Can I get some input on what I am missing here?

**Comment by supervoid0** [comment, marked as answer]
The token you are using probably doesn't have access to the repository or certain action on it. Check if it has the required scopes for Oauth. 
Check this out. - https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps
